src/v1/device_manager_client.ts

```diff
@@ -2580,49 +2623,17 @@ export class DeviceManagerClient {
         },
         res => {
           let data = '';
-          const chunks: any[] = [];
           res.on('data', chunk => (data += chunk));
           res.on('end', () => {
-            let array: [
-              protos.google.cloud.iot.v1.IDeviceRegistry[],
-              protos.google.cloud.iot.v1.IListDeviceRegistriesRequest | null,
-              protos.google.cloud.iot.v1.IListDeviceRegistriesResponse
-            ];
-
-            const ideviceregistry: protos.google.cloud.iot.v1.IDeviceRegistry[] =
-              [];
-
-            //for loop fetching JSON
-            const registry: protos.google.cloud.iot.v1.IDeviceRegistry = {};
-
-            const registriesList = JSON.parse(data);
-            for (const index in registriesList.deviceRegistries) {
-              registry.name = registriesList.deviceRegistries[index].name;
-              registry.id = registriesList.deviceRegistries[index].id;
-              ideviceregistry.push(registry);
-            }
-
-            const ilistDeviceregistriesrequest: protos.google.cloud.iot.v1.IListDeviceRegistriesRequest =
-              {};
-            const ilistdeviceregistriesresponse: protos.google.cloud.iot.v1.IListDeviceRegistriesResponse =
-              {};
-            ilistdeviceregistriesresponse.deviceRegistries = ideviceregistry;
-            // eslint-disable-next-line prefer-const
-            array = [
-              ideviceregistry,
-              ilistDeviceregistriesrequest,
-              ilistdeviceregistriesresponse,
-            ];
-            resolve(array);
+            const deviceRegistries: protos.google.cloud.iot.v1.IDeviceRegistry[] =
```

Comment:
You need to access the deviceRegistries property from the parsed data. This should be 
```
const deviceRegistries: protos.google.cloud.iot.v1.IDeviceRegistry[] = JSON.parse(data).deviceRegistries;
```